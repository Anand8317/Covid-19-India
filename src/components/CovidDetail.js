import React from 'react';
import { AiFillSetting } from 'react-icons/ai';
import { BsFillArrowLeftSquareFill, BsFillMicFill } from 'react-icons/bs';
import { useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { FiArrowRightCircle } from 'react-icons/fi';

const CovidDetail = () => {
  const covid = useSelector((state) => state.covid);
  const covidArr = Object.keys(covid).map((key) => ({ id: key, ...covid[key] }));
  const params = useParams();
  const covidSpe = covidArr.find((c) => c.id === params.title);

  return (
    <section className="detail-section">
      <div className="toolbar">
        <Link to="/" className="toolbar-link">
          <BsFillArrowLeftSquareFill />
        </Link>
        <h1 className="title">
          <Link to="/" className="toolbar-link">Covid</Link>
        </h1>
        <div className="d-row toolbar-link">
          <BsFillMicFill />
          <AiFillSetting />
        </div>
      </div>
      <div className="detail-title">
        <h2>{covidSpe.id}</h2>
      </div>
      <div className="detail-container">
        <div className="detail-header">
          <h3>DETAIL INFORMATION</h3>
        </div>
        <div className="detail-item">
          <h3>Confirmed</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.confirmed}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
        <div className="detail-item">
          <h3>Deceased</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.deceased ? covidSpe.delta7.deceased : 0}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
        <div className="detail-item">
          <h3>Recovered</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.recovered}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
        <div className="detail-item">
          <h3>Tested</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.tested}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
        <div className="detail-item">
          <h3>Vaccinated1</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.vaccinated1}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
        <div className="detail-item">
          <h3>Vaccinated2</h3>
          <div className="arrow-container-detail">
            <p>{covidSpe.delta7.vaccinated2}</p>
            <FiArrowRightCircle className="arrow" />
          </div>
        </div>
      </div>
    </section>
  );
};

export default CovidDetail;
