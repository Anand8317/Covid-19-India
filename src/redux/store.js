import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { covidReducer } from './covid/covid';

const store = configureStore(
  {
    reducer: {
      covid: covidReducer,
    },
  },
  applyMiddleware(thunk),
);
export default store;
