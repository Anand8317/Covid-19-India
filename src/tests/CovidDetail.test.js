import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CovidDetail from '../components/CovidDetail';
import store from '../redux/store';

const App = () => {
  <Provider store={store}>
    <CovidDetail />
  </Provider>;
};

describe('CovidDetail component test', () => {
  it('Testing covid detail page rendering', () => {
    const CovidDetailComponent = renderer.create(<App />).toJSON();
    expect(CovidDetailComponent).toMatchSnapshot();
  });
});
