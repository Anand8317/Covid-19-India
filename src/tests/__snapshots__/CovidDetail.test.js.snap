// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Covid component test Testing covid page rendering 1`] = `null`;

exports[`CovidDetail component test Testing covid detail page rendering 1`] = `null`;
